examples/routing/mini_gamma_IRF.jl

```diff
@@ -30,35 +30,57 @@ forcing_timeseries_dir =
     joinpath(data_file_path, "timeseries", "timeseries_lv05")
 output_dir =
     joinpath(data_file_path, "simulations", "simulations_lv05", "gamma_IRF")
-@info "creating output path"
+@info "output path: $output_dir"
 if !isdir(output_dir)
     mkpath(output_dir)
 end
 
+# data information
+data_start_date = Date("1996-01-01", "yyyy-mm-dd")
+data_end_date = Date("2014-12-31", "yyyy-mm-dd") # of entire simulation
+data_step = Day(1)
+data_date_window = DateWindow(
+    start_date = data_start_date,
+    end_date = data_end_date,
+    date_step = data_step,
+)
+
 # build environment
 env = Environment(
     basin_ids_file = basin_ids_file,
     attributes_file = attributes_file,
     graph_file = graph_file,
     forcing_timeseries_dir = forcing_timeseries_dir,
+    date_window = data_date_window,
     output_dir = output_dir,
     forcing_timeseries_file_prefix = "basin_",
 )
 
-## evolutionary model, evolving a state over time
-model_types = ["instant"]
-model_type = model_types[1]
-
-start_date = Date("1996-01-01", "yyyy-mm-dd")
-end_date = Date("2014-12-31", "yyyy-mm-dd")
-
 # River state loaded into csv files currently, placehodler variable
-hillslope_data = zeros(10, 10)
-channel_data = zeros(10, 10)
-
-river_state = RiverState(hillslope_data, channel_data)
+history_length = 50 * Day(1)
```

Comment:
Shouldnt this be equal to t_max_hillslope?

Reply:
I have left them decoupled for now, internally taking
```
minimum(history_length, ChannelModle.t_max)
```
I'll check out if coupling it, which is sensible.